lib/compiler/assets-manager.ts

```diff
@@ -96,6 +78,10 @@ export class AssetsManager {
           .on('change', (path: string) => this.actionOnFile({ ...option, path, action: 'change' }))
           .on('unlink', (path: string) => this.actionOnFile({ ...option, path, action: 'unlink' }));
 
+        if (!isWatchEnabled) {
+          watcher.on('ready', () => watcher.close());
```

Comment:
Same here https://github.com/nestjs/nest-cli/pull/1829/files#r1017657410

Reply:
Same here. Watcher event listener return type is void, so it will not listen to the promise.